### PR TITLE
Add more payment methods + webhook + best practices

### PR DESCRIPTION
**Pull request by danfowler**
* Add a more fully fleshed out example store incl. webhook support, sources support, best practices for optional fields, and charging customers

### CODE REVIEW

File: payment/checkout.py [hunk 1]
```diff
@@ -1,70 +1,159 @@
...
             currency=current_app.config.get("STORE_CURRENCY"),
-            card=token,
-            metadata={"app": "Omise Flask", "cart": {
-                "items": cart.items()}, "order_id": str(order_id)},
+            metadata={
+                "app": "Omise Flask",
+                "cart": {"items": cart.items()},
+                "order_id": str(order_id),
+            },
+            return_uri=f"https://omise-flask-example.herokuapp.com/orders/{order_id}/complete",
```
Comment (danfowler) [inline payment/checkout.py]: Need to change to environment variable.

File: payment/checkout.py [hunk 2]
```diff
@@ -1,70 +1,159 @@
...
+
+
+def processed(chrg, already_redirected=False):
+    """
+    Process charge depending on payment method.  `already_redirected`
+    tries to ensure that the customer is redirected only once.
+    """
+
+    cart = Cart()
+    order_id = chrg._attributes["metadata"]["order_id"]
```
Comment (danfowler) [inline payment/checkout.py:48]: Is there a better way to access these attributes?

File: payment/checkout.py [hunk 3]
```diff
@@ -1,70 +1,159 @@
...
 from store.cart import Cart, Price
-import uuid
 
 checkout = Blueprint("checkout", __name__, template_folder="templates")
 
 
+def get_client_ip():
+    if "X-Forwarded-For" in request.headers:
+        return (
+            next(iter(request.headers["X-Forwarded-For"].split(",")))
```
Comment (danfowler) [inline payment/checkout.py]: Is this specific to Heroku?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.mypy_cache
```
Comment (danfowler) [inline .gitignore:1]: Need to update README.md

File: payment/templates/barcode.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{% extends "layout.html" %}
```
Comment (danfowler) [inline payment/templates/barcode.html:1]: Should this be able to display more than one type of barcode?

File: config.py [hunk 6]
```diff
@@ -8,3 +8,4 @@ class Config(object):
     SECRET_KEY = os.environ["FLASK_SECRET_KEY"]
     STORE_LOCALE = os.environ["STORE_LOCALE"] or "th_TH"
     STORE_CURRENCY = os.environ["STORE_CURRENCY"] or "THB"
+    PREFERRED_URL_SCHEME = os.environ["PREFERRED_URL_SCHEME"] or "https"
```
Comment (vernomcrp) [inline config.py]: IMHO, I think we can do this `os.environ.get("PREFERRED_URL_SCHEME", "https")`

File: payment/checkout.py (hunk 2 above)
Comment (vernomcrp) [reply payment/checkout.py:48]: @danfowler , Do you have any concern about this way ?

File: config.py (hunk 6 above)
Comment (danfowler) [reply config.py]: 👍 Great, thanks!

Comment (danfowler) [reply config.py]: Fixing

File: payment/templates/barcode.html (hunk 5 above)
Comment (danfowler) [reply payment/templates/barcode.html:1]: Nah

**Pull request merged**